Adapter/PhpcrOdmAdapter.php

```diff
@@ -137,12 +138,24 @@ public function createAutoRoute(UriContext $uriContext, $contentDocument, $autoR
             $parentDocument = $document;
         }
 
+        $finalAutoRoutePath = $path . '/' . $headName;
+        $node = $this->dm->find(null, $finalAutoRoutePath);
+        if ($node) {
+            if ($node instanceof Generic) {
+                return $this->convertGenericNodeInAutoRouteNode($node, $contentDocument, $autoRouteTag, $routeType);
+            }
+            $nodeClass = get_class($node);
+            $genericFqcn = 'Doctrine\ODM\PHPCR\Document\Generic';
+            throw new \RuntimeException(
+                "Unexpected node class '$nodeClass' at path '$finalAutoRoutePath'. Only '$genericFqcn' expected."
+            );
+        }
         $headRoute = new $this->autoRouteFqcn();
         $headRoute->setContent($contentDocument);
         $headRoute->setName($headName);
         $headRoute->setParent($document);
         $headRoute->setAutoRouteTag($autoRouteTag);
-        $headRoute->setType(AutoRouteInterface::TYPE_PRIMARY);
+        $headRoute->setType($routeType);
```

Comment:
Revert this .. we only ever set this to TYPE_PRIMARY. Making this dynamic is just confusing?